src/encoded/static/components/report.js

```diff
@@ -411,19 +552,14 @@ class Report extends React.Component {
         this.context.navigate(url.format(parsedUrl));
     }
 
-    toggleColumn(toggledPath) {
+    setColumnState(newColumns) {
+        // Gets called when the user clicks the Select button in the ColumnSelector modal.
+        // `newColumns` has the same format as `columns` returned from `columnChoices`, but
+        // `newColumn s`has the user's chosen columns from the modal, while `columns` has the
+        // columns selected by the query string.
         const parsedUrl = url.parse(this.context.location_href, true);
-        const type = parsedUrl.query.type;
-        const schema = this.props.schemas[type];
-        const queryFields = parsedUrl.query.field ? (typeof parsedUrl.query.field === 'object' ? parsedUrl.query.field : [parsedUrl.query.field]) : undefined;
-        const columns = columnChoices(schema, queryFields);
 
-        const fields = [];
-        _.mapObject(columns, (column, path) => {
-            if (path === toggledPath ? !column.visible : column.visible) {
-                fields.push(path);
-            }
-        });
+        const fields = Object.keys(newColumns).filter(columnPath => newColumns[columnPath].visible);
```

Comment:
Do we need the fields variable since we're now just putting it directly into parsedUrl.query.field?

Reply:
We could. Does this need to change? It’s zero cost as far as Javascript is concerned.

Reply:
No it's not vital so long as we keep it mind in the future.